opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/sitecustomize.py

```diff
@@ -32,24 +33,40 @@
 
 
 def _load_distros() -> BaseDistro:
-    for entry_point in iter_entry_points("opentelemetry_distro"):
+    entry_points = iter_entry_points("opentelemetry_distro")
+    loaded_distro = None
+    loaded_distro_name = None
+    for entry_point in entry_points:
```

Comment:
Using a loop for this seems weird. Can we run a check for length instead? Check if `len(entry_points)` is > `1` and log the error. Also use `entry_points[0]` to get reference to the first found distro and guard it for `IndexError` exception.

Reply:
I agree it's a little weird 😓 I took the tradeoff though because the `entry_points` item is a generator so you can't run `len(entry_points)` on it 😕 

I guess I didn't want to run the cost of iterating through the iterator until we know for certain that there was > 1 distro installed, but I'll think of a re-write to make this cleaner 🙂

Reply:
how about this? :D
```
try
  distro = entry_points.__next__()
except StopIteration:
  # no distro found
```

It doesn't look clean either. I'm fine with loop or using __next__. Just that loop does not clarify the intention of this code block until you read it thoroughly. 

`list(entry_points)[0]` would probably be the best thing in terms of readability but not sure about performance as it might hit the disk on every iteration. Then again, ideally no one should have more than one distros installed and if they have, they probably are going to get undefined and inconsistent behavior anyway.

Reply:
I went with a similar solution using `next()` on the latest commit so hopefully it looks cleaner! I'll resolve this comment and we can continue there 🙂

Reply:
Sorry, just reviewed it in more detail. We are enforcing this by raising a RuntimeError if more than one distros are found. I think in that case startup performance shouldn't be a concern since the service will fail anyway. So I think converting to a list and running checks on length should be fine.